modules/ROOT/attachments/admin-api.yaml

```diff
@@ -809,6 +809,155 @@ paths:
         200:
           description: Stop maintenance success
           content: {}
+  /v1/migrations:
+    get: 
+      summary: List all available migrations
+      operationId: list_migrations
+      responses: 
+        200:
+          description: OK
+          content: 
+            application/json:
+              {}
+      tags: 
+        - Migrations
+    put: 
+      summary: Create migration
+      operationId: add_migration
+      tags: 
+        - Migrations
+      requestBody: 
+        description: Migration configuration
+        required: true
+        content: 
+          application/json:
+            schema:                 
+              oneOf: 
+              - $ref: "#/components/schemas/inbound_migration"
+              - $ref: "#/components/schemas/outbound_migration"
+              discriminator: 
+                propertyName: migration_type
+                mapping: 
+                  inbound: "#/components/schemas/inbound_migration"
+                  outbound: "#/components/schemas/outbound_migration"
+      responses: 
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:                 
+                oneOf: 
+                - $ref: "#/components/schemas/inbound_migration"
+                - $ref: "#/components/schemas/outbound_migration"
+  /v1/migrations/{id}:
+    get:
+      summary: Get migration state
+      operationId: get_migration
+      tags: 
+        - Migrations
+      parameters: 
+        - name: id
+          in: path 
+          required: true
+          schema: 
+            type: integer
+            format: int32  
+      responses: 
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:                 
+                oneOf: 
+                - $ref: "#/components/schemas/inbound_migration"
+                - $ref: "#/components/schemas/outbound_migration"
+    post: 
+      summary: Execute migration action
+      operationId: execute_migration_action
+      tags: 
+        - Migrations
+      parameters: 
+        - name: id 
+          in: path
+          required: true
+          schema: 
+            type: integer
+            format: int32
+        - name: action
+          in: query
+          required: true
+          description: Action to execute for migration
+          schema: 
+            type: string
+            enum: 
+            - prepare
+            - execute
+            - finish
+            - cancel  
+      responses: 
+        200:
+          description: OK
+          content: {}
+    delete:
```

Comment:
There is no example, schema or description for this endpoint.

Reply:
@asimms41 same here in that there is no request body to include, just the migration ID in the path.

Reply:
Is there any way we can add help text/labelling?